components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/CurrencyInputView.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.components.compose.molecule.input
+
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.KeyboardType
+
+@Composable
+fun CurrencyInputView(
+    modifier: Modifier = Modifier,
+    onInputValueChange: ((String) -> Unit)? = null,
+    placeholderText: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    singleLine: Boolean = false,
+    characterCount: Int? = null,
+    enableDecimal: Boolean = true,
+) {
+
+    // pattern matches a decimal number
+    val pattern = remember { Regex("^[0-9]+(\\.?)([0-9]?[0-9]?)") }
+
+    fun enableDecimal(input: String, localValue: String): (String) {
```

Comment:
I think it's a bit confusing having this function named the same as the boolean val in the parent function. What is this function actually doing? 'Enable' implies a toggle or boolean I think